### PR TITLE
Add definitions to ChatContainer’s getMessages interface

### PR DESCRIPTION
**Pull request by howawong**
- function getMessages  accepts beforeMessage or beforeMessageId

Connects #140

### CODE REVIEW

File: chat/src/main/java/io/skygear/plugins/chat/CacheController.java [hunk 1]
```diff
@@ -20,22 +20,21 @@
...
 import io.skygear.skygear.Error;
 
 import static io.skygear.plugins.chat.MessageCacheObject.KEY_CONVERSATION_ID;
 import static io.skygear.plugins.chat.MessageCacheObject.KEY_CREATION_DATE;
 import static io.skygear.plugins.chat.MessageCacheObject.KEY_DELETED;
 import static io.skygear.plugins.chat.MessageCacheObject.KEY_EDITION_DATE;
-import static io.skygear.plugins.chat.MessageCacheObject.KEY_SEND_DATE;
+import static io.skygear.plugins.chat.MessageCacheObject.KEY_RECORD_ID;
 
+import static io.skygear.plugins.chat.MessageCacheObject.KEY_SEQ;
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/CacheController.java]: `KEY_SEQUENCE` should be better

File: chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java [hunk 2]
```diff
@@ -627,10 +627,55 @@ public void onLambdaFail(Error error) {
...
+        args.put("before_message_id", beforeMessageId);
+        args.put("order", order);
+        getMessages(args, callback);
+
+    }
+
     /* --- Message --- */
     /**
      * Gets messages.
-     *
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java]: don’t need to remove this line

File: chat/src/main/java/io/skygear/plugins/chat/Message.java [hunk 3]
```diff
@@ -179,6 +180,17 @@ public Date getSendDate() {
         return this.sendDate;
     }
 
+    /**
+     * Gets seq.
+     *
+     * @return the seq
+     */
+    @NonNull
+    public int getSeq() {
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/Message.java]: s/getSeq/getSequence/

File: chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java [hunk 4]
```diff
@@ -659,15 +716,12 @@ public void onSuccess(@Nullable List<Message> object) {
 
             @Override
             public void onFail(@NonNull Error error) {
-
+                Log.e(TAG, "Failed to load message from cache: " + error.getMessage());
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java]: 👍

File: chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java [hunk 5]
```diff
@@ -627,10 +627,55 @@ public void onLambdaFail(Error error) {
...
+     * Gets messages.
+     * @param conversation     the conversation
+     * @param limit            the limit
+     * @param beforeMessageId  the before message
+     * @param order            the order, either 'edited_at' or '_created_at'
+     * @param callback         the callback
+     */
+    public void getMessages(@NonNull final Conversation conversation,
+                            final int limit,
+                            @Nullable final String beforeMessageId,
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java]: here `beforeMessageId` is okay

**Comment by howawong** [comment]
@cheungpat  added migration

**Pull request merged**